src/console-app-device-code/Program.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Microsoft.Identity.Client;
+using System.Net.Http.Headers;
+using System.Text.Json;
+
+var config = new PublicClientApplicationOptions
+{
+    // 'Directory (tenant) ID' of app registration in the Azure portal - this value is a GUID
+    TenantId = "",
+
+    // 'Application (client) ID' of app registration in Azure portal - this value is a GUID
+    ClientId = ""
+};
+
+// In order to take advantage of token caching, your MSAL client singleton must
+// have a lifecycle that at least matches the lifecycle of the user's session in
+// the application.
+IPublicClientApplication publicMsalClient = PublicClientApplicationBuilder.CreateWithApplicationOptions(config)
+                                                                          .Build();
+
+AuthenticationResult? msalAuthenticationResult = null;
+
+// Ideally, you'd first attempt to used a cached access token if one was available. This will renew 

```

Comment:
from what I reading the pattern is never used in our sample. But more importantly, Is this something we want to teach our users to incorporate within their console apps as part of this tutorial? I think one way to think about this is whether we are envisioning this is going to be part of the docs.

If we eliminate this, we simplify this a lot.

Reply:
That is correct, it's not used in this sample workflow (would need to reorganize the code for it to show utility).  I feel weird not teaching the tenants of token caching here, but I also just noticed that the Python and PowerShell ones also do not do this.  So, you're right to identify that this is very much different than the other two.  This is where I'd lean on content dev to say, "what do you plan to teach?" (I'd be shocked if they don't plan to teach the basics of client re-use for token caching purposes.... but I could be very wrong.)

I wonder if this can be addressed via a strategically placed comment instead of code. Or a mention of it in the README's section that talks about the implementation.

I also wonder if leaving it in is okay, because when docs goes to "snippet this" the snippets could easily "cut" this part out of the flow, as nothing really "depends on" that part of the code existing (no variables are defined in there, etc).

Hum.  Now I'm torn on what to do.

I'm thinking maybe just fall back to "consistency" and follow suit with what we did on python and powershell, but augement that a bit with README and/or comment contents.  What do you think @ferantivero and @skabou?

Reply:
I did notice that it was different than what was done with the PowerShell and python tutorials.  It "seems" safer to include it than not, but it depends what the objective of Content Dev is.  Likewise I'm conflicted.  I recall with other tutorials being advised to make them more concise and simple, but including logic for caching may be something that should be covered.  I don't really have a firm answer.

Reply:
Updated in https://github.com/Azure-Samples/ms-identity-docs-code-dotnet/pull/24/commits/aa6b6643c03f09e27ea445683083d5d049e8c46a -- removed the code, and replaced it with a couple of mentions.

Reply:
@ckittel I didn't notice that we're not doing token cache checking in the Psh and Py samples, but all samples should avoid the anti-pattern of not first checking the cache for a token.

See https://github.com/Azure-Samples/ms-identity-docs-code-swift/pull/1#discussion_r825219123 for more of an explanation.